File: Competitive Programming 3/Section 1.4/Games (Others), Easier/11459 - Snakes and Ladders/C++/main.cpp
#include<bits/stdc++.h>

using namespace std;

int t,a,b,c,goDif[100];

void printAns(int * players){
    for(int i=0;i<a;i++)
        cout<<"Position of player "<<i+1<<" is " <<players[i]+1<<".\n";
}
int main(){
    ios_base::sync_with_stdio(0);cin.tie(0);
    cin>>t;
    while(t--){
        cin>>a>>b>>c;
        int players[a],orig,dest,val;
        memset(players,0,sizeof(players));
        memset(goDif,0,sizeof(goDif));
        for(int i=0;i<b;i++){
            cin>>orig>>dest;
            goDif[orig-1]=dest-orig;   
        }
        bool won=false;
        for(int i=0,player=0;i<c;i++){
            cin>>val;
            if(!won){
                players[player]=min(players[player]+val,99);
                players[player]+=goDif[players[player]];
                if(players[player]==99){//print answer
                    won=true;
                    printAns(players);
                }
                player=(player+1)%a;
            }
        }
        if(!won)
            printAns(players);
    }
    return 0;
}

File: Competitive Programming 3/Section 1.4/Games (Others), Easier/11459 - Snakes and Ladders/Python/main.py
def printAns(players):
    for i,p in enumerate(players,1):
        print('Position of player '+str(i)+' is '+str(p+1)+'.')
def main():
    t=int(input())
    for _ in range(t):
        a,b,c=map(int,input().split())
        players,goto=[0]*a,[-1]*100 #0 to 99
        for _ in range(b):
            origin,dest=map(int,input().split())
            goto[origin-1]=dest-1
        won=False
        for player,_ in enumerate(range(c)):
            val=input()
            if not won:
                player%=a 
                players[player]=min(players[player]+int(val),99)
                if goto[players[player]]!=-1:
                    players[player]=goto[players[player]]
                if players[player] == 99:#print answer
                    won=True
                    printAns(players)
        if not won:
            printAns(players)
main()